app/models/test.rb

```diff
@@ -1,17 +1,20 @@
 class Test < ApplicationRecord
-  belongs_to :category, optional: true                         # тест имеет одну категорию,
-                                                               # наличие связанных объектов не валидируется
-  has_many :questions, dependent: :destroy                     # у одного теста может быть много вопросов
+  belongs_to :category, optional: true
+  has_many :questions, dependent: :destroy
 
-  has_many :selected_tests, dependent: :destroy                # один тест могут проходит несколько пользователей
+  has_many :selected_tests, dependent: :destroy
   has_many :users, through: :selected_tests
 
-  belongs_to :author, class_name: "User", dependent: :destroy  # тест принадлежит одному автору
+  belongs_to :author, class_name: "User", dependent: :destroy
 
-	def self.tests_titles(category)
-    Test.joins(:category)
-        .where(categories: {title: category})
-        .order(id: :desc)
-        .pluck(:title)
-  end
+  scope :easy, -> { where(level: 0..1) }
+  scope :middle, -> { where(level: 2..4) }
+  scope :hard, -> { where(level: 5..Float::INFINITY) }
+
+  scope :tests_titles, -> (category) {joins(:category).where(categories: {title: category}).order(id: :desc).pluck(:title)}
+
+  validates :level, numericality: { only_integer: true, 
+                                    greater_than_or_equal_to: 0 }
+  validates :title, presence: true, uniqueness: { scope: :level,
```

Comment:
не обязательно к выполнению
для информации
[в документации](https://guides.rubyonrails.org/active_record_validations.html#uniqueness) к хелперу `uniqueness` сказано следующее:
> It does not create a uniqueness constraint in the database, so it may happen that two different database connections create two records with the same value for a column that you intend to be unique. To avoid that, you must create a unique index on that column in your database.

другими словами, чтобы эта была настоящая уникальность (не только на уровне приложения, но и на уровне базы), нужно в базе создать уникальный индекс
про правильную миграцию можно посмотреть к примеру на [СО](https://stackoverflow.com/questions/32927262/create-migration-with-uniqueness-to-a-scope)